src/main/java/com/thomashan/coup/action/PerformAction.java

```diff
@@ -0,0 +1,53 @@
+package com.thomashan.coup.action;
+
+import com.thomashan.coup.player.Player;
+
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+
+public class PerformAction implements Action {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A class which only has private constructors should be final](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)